frontend/src/contexts/__tests__/jest/AuthenticationContext.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  AuthenticationContextType,
+  deserializeAuthenticationContext,
+  isStudentLocallyAuthenticated,
+  serializeAuthenticationContext,
+} from "@/contexts/AuthenticationContext";
+import { UserRole } from "@/i18n/user-role-messages";
+import KeyPair from "@/utilities/crypto/KeyPair";
+import { subtle } from "crypto";
+import {
+  getAuthenticatedNonStudentTeacherContext,
+  getAuthenticatedTeacherContext,
+  getFullyAuthenticatedStudentContext,
+  getLocallyAuthenticatedStudentContext,
+  getUnauthenticatedContext,
+} from "../mock-contexts";
+
+const crypto = subtle as SubtleCrypto;
+
+describe("AuthenticationContext", () => {
+  describe("isStudentLocallyAuthenticated", () => {
+    it.each([["student (locally)", getLocallyAuthenticatedStudentContext]])(
+      "should return true if the user is authenticated as %s",
+      async (_, getContext) => {
+        expect(isStudentLocallyAuthenticated(await getContext())).toBe(true);
+      },
+    );
+
+    it.each([
+      ["nobody", getUnauthenticatedContext],
+      ["student (fully)", getFullyAuthenticatedStudentContext],
+      ["teacher", getAuthenticatedTeacherContext],
+      ["admin", getAuthenticatedNonStudentTeacherContext(UserRole.admin)],
```

Comment:
this makes me think that we should just call this `getAuthenticatedAdminContext`